test/e2e/framework/expect.go

```diff
@@ -43,7 +46,22 @@ func ExpectNoError(err error, explain ...interface{}) {
 // ExpectNoErrorWithOffset checks if "err" is set, and if so, fails assertion while logging the error at "offset" levels above its caller
 // (for example, for call chain f -> g -> ExpectNoErrorWithOffset(1, ...) error would be logged for "f").
 func ExpectNoErrorWithOffset(offset int, err error, explain ...interface{}) {
-	gomega.ExpectWithOffset(1+offset, err).NotTo(gomega.HaveOccurred(), explain...)
+	if err == nil {
+		return
+	}
+
+	// Errors usually contain unexported fields. We have to use
+	// a formatter here which can print those.
+	Logf("Unexpected error:\n%s", format.Object(err, 1))
+	prefix := ""
+	if len(explain) > 0 {
+		if str, ok := explain[0].(string); ok {
+			prefix = fmt.Sprintf(str, explain[1:]...) + ": "
+		} else {
+			prefix = fmt.Sprintf("unexpected explain arguments, need format string: %v", explain)
+		}
+	}
+	Fail(prefix+err.Error(), 1+offset)
```

Comment:
Consider adding a comment here as to why we're not using the default gomega assertions.

Reply:
Done.